### PR TITLE
DApp-1840 table fixes

### PR DESCRIPTION
**Pull request by rohitmalhotra1420**
### Ticket Number

- [1840](https://github.com/push-protocol/push-dapp/issues/1840)

### Description

- **Problem/Feature**:

- Added support for row click in table
- Added support for loading state 
- Support for null state
- Support for failure state

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other (please describe):

### Checklist

- [ ] **Quick PR**: Is this a quick PR? Can be approved before finishing a coffee.
  - [ ] Quick PR label added
- [ ] **Not Merge Ready**: Is this PR dependent on some other PR/tasks and not ready to be merged right now.
  - [ ] DO NOT Merge PR label added

### Frontend Guidelines

- [x] Followed frontend guidelines as per the [Guidelines Notion Doc](https://www.notion.so/pushprotocol/Frontend-dApp-Guidelines-1d7806ae3d9e4569a340b563dcd0536c)

### Build & Testing

- [x] No errors in the build terminal
- [x] Engineer has tested the changes on their local environment
- [ ] Engineer has tested the changes on deploy preview

### Screenshots/Video with Explanation

- **Before:** Explain the previous behavior

- **After:** What's changed now

### Review & Approvals

- [x] Self-review completed
- [x] Code review by at least one other engineer
- [x] Documentation updates if applicable

### CODE REVIEW

File: src/blocks/table/Table.tsx [hunk 1]
```diff
@@ -116,59 +169,111 @@ const Table: FC<TableProps> = ({ backgroundColor = 'surface-secondary', columns,
...
-                key={record.id}
-                item={record}
+    <TableContainer>
+      {loading && !error && (
+        <OverlayContainer blur>
+          <Spinner
+            size="medium"
+            variant="primary"
+          />
+          <Text
```
Comment (mishramonalisha76) [inline src/blocks/table/Table.tsx:185]: I think we shouldnt using other block components right ?

Comment (mishramonalisha76) [reply src/blocks/table/Table.tsx:185]: Text can be a styled component here itself

File: src/blocks/table/Table.tsx [hunk 2]
```diff
@@ -116,59 +169,111 @@ const Table: FC<TableProps> = ({ backgroundColor = 'surface-secondary', columns,
...
-            ))}
-          </Body>
-        </>
+                {error
+                  ? 'Please try again in a few minutes or reload the page.'
+                  : 'Try adjusting your search or filter to find what you’re looking for'}
+              </Text>
+            </NullStateTextContiner>
+            {error && onRetry && (
+              <Button
```
Comment (mishramonalisha76) [inline src/blocks/table/Table.tsx:214]: Are we supposed to use other block components?

**Review by mishramonalisha76 — CHANGES_REQUESTED** [review]

File: src/blocks/table/Table.tsx [hunk 3]
```diff
@@ -45,7 +54,64 @@ const StyledRowCell = styled(Cell)<{ cellAlignment?: Column['cellAlignment'] }>`
...
+const NullStateHeading = styled.span`
+  ${() => getTextVariantStyles('bm-semibold', 'text-primary')}
+`;
+
+const NullStateDescription = styled.span`
+  ${() => getTextVariantStyles('bes-regular', 'text-tertiary')}
+`;
+
+const LoadingText = styled.span`
+  ${() => getTextVariantStyles('bm-semibold', 'text-tertiary')}
```
Comment (mishramonalisha76) [inline src/blocks/table/Table.tsx:95]: you can keep one text styled component with props to pass the details

**Review by mishramonalisha76 — CHANGES_REQUESTED** [review]

File: src/blocks/table/Table.tsx (hunk 3 above)
Comment (rohitmalhotra1420) [reply src/blocks/table/Table.tsx:95]: I did this intentionally
- To keep it simple and 
- To allow me to add additional css to specific text in future.

Comment (mishramonalisha76) [reply src/blocks/table/Table.tsx:95]: okay

**Review by mishramonalisha76 — APPROVED** [review]

**Pull request merged**